test/correctness/math.cpp

```diff
@@ -70,136 +69,153 @@ bool relatively_equal(value_t a, value_t b, Target target) {
-uint16_t absd(int16_t a, int16_t b) { return a < b ? b - a : a - b; }
-uint32_t absd(int32_t a, int32_t b) { return a < b ? b - a : a - b; }
-uint8_t absd(uint8_t a, uint8_t b) { return a < b ? b - a : a - b; }
-uint16_t absd(uint16_t a, uint16_t b) { return a < b ? b - a : a - b; }
-uint32_t absd(uint32_t a, uint32_t b) { return a < b ? b - a : a - b; }
-
+float absd(float a, float b) {
+    return a < b ? b - a : a - b;
+}
+double absd(double a, double b) {
+    return a < b ? b - a : a - b;
+}
+uint8_t absd(int8_t a, int8_t b) {
+    return a < b ? b - a : a - b;
+}
+uint16_t absd(int16_t a, int16_t b) {
+    return a < b ? b - a : a - b;
+}
+uint32_t absd(int32_t a, int32_t b) {
+    return a < b ? b - a : a - b;
+}
+uint8_t absd(uint8_t a, uint8_t b) {
+    return a < b ? b - a : a - b;
+}
+uint16_t absd(uint16_t a, uint16_t b) {
+    return a < b ? b - a : a - b;
+}
+uint32_t absd(uint32_t a, uint32_t b) {
+    return a < b ? b - a : a - b;
+}
 
 // Using macros to expand name as both a C function and an Expr fragment.
 // It may well be possible to do this without macros, but that is left
 // for another day.
 
 // Version for a one argument function.
-#define fun_1(type_ret, type, name, c_name)                             \
-    void test_##type##_##name(Buffer<type> in) {                        \
-        Target target = get_jit_target_from_environment();              \
-        if (!target.supports_type(type_of<type>())) {                   \
-            return;                                                     \
-        }                                                               \
-        Func test_##name("test_" #name);                                \
-        Var x("x"), xi("xi");                                           \
-        test_##name(x) = name(in(x));                                   \
-        if (target.has_gpu_feature()) {                                 \
-            test_##name.gpu_tile(x, xi, 8);                             \
-        } if (target.has_feature(Target::OpenGLCompute)) {              \
-            test_##name.gpu_tile(x, xi, 8, TailStrategy::Auto,          \
-                                 DeviceAPI::OpenGLCompute);             \
-        } else if (target.features_any_of({Target::HVX_64, Target::HVX_128})) { \
-            test_##name.hexagon();                                      \
-        }                                                               \
-        Buffer<type_ret> result = test_##name.realize(in.extent(0), target); \
-        for (int i = 0; i < in.extent(0); i++) {                        \
-            type_ret c_result = c_name(in(i));                          \
-            if (!relatively_equal(c_result, result(i), target)) {       \
+#define fun_1(type_ret, type, name, c_name)                                                                                                                                   \
```

Comment:
trailing backslash is a long way to the right

btw forgive any typos in these notes. Spilled coffee on keyboard and my letter i is flaky.